src/app/Agent/AgentProbeTemplates.tsx

```diff
@@ -0,0 +1,442 @@
+import { LoadingView } from '@app/LoadingView/LoadingView';
+import { authFailMessage, ErrorView, isAuthFail } from '@app/ErrorView/ErrorView';
+import { ProbeTemplate } from '@app/Shared/Services/Api.service';
+import { DeleteWarningType } from '@app/Modal/DeleteWarningUtils';
+import { DeleteWarningModal } from '@app/Modal/DeleteWarningModal';
+
+export const AgentProbeTemplates = () => {
+  const context = React.useContext(ServiceContext);
+  const history = useHistory();
+
+  const [templates, setTemplates] = React.useState([] as ProbeTemplate[]);
+  const [filteredTemplates, setFilteredTemplates] = React.useState([] as ProbeTemplate[]);
+  const [filterText, setFilterText] = React.useState('');
+  const [modalOpen, setModalOpen] = React.useState(false);
+  const [uploadFile, setUploadFile] = React.useState(undefined as File | undefined);
+  const [uploadFilename, setUploadFilename] = React.useState('');
+  const [uploading, setUploading] = React.useState(false);
+  const [fileRejected, setFileRejected] = React.useState(false);
+  const [sortBy, setSortBy] = React.useState({} as ISortBy);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [rowDeleteData, setRowDeleteData] = React.useState({} as IRowData);
+  const [warningModalOpen, setWarningModalOpen] = React.useState(false);
+  const addSubscription = useSubscriptions();
+
+  const tableColumns = React.useMemo(
+    () => [
+      { title: 'name', transforms: [sortable] },
+      { title: 'xml', transforms: [sortable] },
+    ],
+    [sortable]
+  );
+
+  React.useEffect(() => {
+    let filtered;
+    if (!filterText) {
+      filtered = templates;
+    } else {
+      const ft = filterText.trim().toLowerCase();
+      filtered = templates.filter(
+        (t: ProbeTemplate) => t.name.toLowerCase().includes(ft) || t.xml.toLowerCase().includes(ft)
+      );
+    }
+    const { index, direction } = sortBy;
+    if (typeof index === 'number') {
+      const keys = ['name', 'xml'];
+      const key = keys[index];
+      const sorted = filtered.sort((a, b) => (a[key] < b[key] ? -1 : a[key] > b[key] ? 1 : 0));
+      filtered = direction === SortByDirection.asc ? sorted : sorted.reverse();
+    }
+    setFilteredTemplates([...filtered]);
+  }, [filterText, templates, sortBy]);
+
+  const handleTemplates = React.useCallback(
+    (templates) => {
+      setTemplates(templates);
+      setIsLoading(false);
+      setErrorMessage('');
+    },
+    [setTemplates, setIsLoading, setErrorMessage]
+  );
+
+  const handleError = React.useCallback(
+    (error) => {
+      setIsLoading(false);
+      setErrorMessage(error.message);
+    },
+    [setIsLoading, setErrorMessage]
+  );
+
+  const refreshTemplates = React.useCallback(() => {
+    setIsLoading(true);
+    addSubscription(
+      context.target
+        .target()
+        .pipe(
+          concatMap((target) => context.api.getProbeTemplates()),
```

Comment:
Seems like api call is not specific to a target. Should we remove `context.target.target`?